### PR TITLE
DEVORTEX-3143 Improve logging

### PR DESCRIPTION
**Pull request by a-shevel**

### CODE REVIEW

File: middleware/auth-utils/grant-manager.js [hunk 1]
```diff
@@ -421,13 +421,19 @@ const fetch = (manager, handler, options, params) => {
 
     const req = getProtocol(options).request(options, (response) => {
       if (response.statusCode < 200 || response.statusCode > 299) {
-        return reject(response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ]);
+        let response = '';
+        response.on('data', (d) => (response += d.toString()));
+        response.on('end', () => {
+          const message = response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ] + ' ' + response;
+          reject(new Error(message));
+        });
```
Comment (dnetrebenko-smartling) [inline middleware/auth-utils/grant-manager.js]: IMO, this change will not work properly. you used variable `response` to store response's body and you  have response object with the same name.

File: middleware/auth-utils/grant-manager.js [hunk 2]
```diff
@@ -421,13 +421,19 @@ const fetch = (manager, handler, options, params) => {
...
+      } else {
+        let json = '';
+        response.on('data', (d) => (json += d.toString()));
+        response.on('end', () => {
```
Comment (dnetrebenko-smartling) [inline middleware/auth-utils/grant-manager.js]: the starting of `else` block is identical to code in `if` block...  IMO, you should re-use it and put
`if (response.statusCode < 200 || response.statusCode > 299)`
to
`response.on('end', () => {`

File: middleware/auth-utils/token.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ function Token (token, clientId) {
       this.signature = new Buffer(parts[2], 'base64');
       this.signed = parts[0] + '.' + parts[1];
     } catch (err) {
+      console.error('Failed to parse JWT token', err);
```
Comment (dnetrebenko-smartling) [inline middleware/auth-utils/token.js]: could  we use log instead of console?

File: middleware/grant-attacher.js [hunk 4]
```diff
@@ -21,6 +21,12 @@ module.exports = function (keycloak) {
       .then(grant => {
         request.kauth.grant = grant;
       })
-      .then(next).catch(() => next());
+        .then(next).catch(err => {
+        // err can be undefined
+        if (err) {
+            console.error('Failed to get grant', err);
```
Comment (dnetrebenko-smartling) [inline middleware/grant-attacher.js:27]: and again: could we avoid to use `console`?

File: middleware/post-auth.js [hunk 5]
```diff
@@ -44,7 +44,7 @@ module.exports = function (keycloak) {
         try {
           keycloak.authenticated(request);
         } catch (err) {
-          console.log(err);
+          console.error(err);
```
Comment (dnetrebenko-smartling) [inline middleware/post-auth.js:47]: and again

File: middleware/auth-utils/grant-manager.js (hunk 1 above)
Comment (a-shevel) [reply middleware/auth-utils/grant-manager.js]: resolved

File: middleware/auth-utils/grant-manager.js (hunk 2 above)
Comment (a-shevel) [reply middleware/auth-utils/grant-manager.js]: resolved

File: middleware/auth-utils/token.js (hunk 3 above)
Comment (a-shevel) [reply middleware/auth-utils/token.js]: I did not want to introduce any additional dependencies in this package. Let's discuss

Comment (dnetrebenko-smartling) [reply middleware/auth-utils/token.js]: Oh... I forgot that it isn't our own library... never mind

File: middleware/auth-utils/grant-manager.js [hunk 6]
```diff
@@ -420,14 +420,20 @@ const fetch = (manager, handler, options, params) => {
...
+          const grantType = typeof params === 'object' ? (params.grant_type || '') : '';
+          const req = `${options.method}:${options.path} ${grantType}`;
+          const resp = `${response.statusCode}: ${http.STATUS_CODES[response.statusCode]} ${json}`;
+          reject(new Error(resp + ' on ' + req));
+        });
+      } else {
+        response.on('end', () => {
+          handler(resolve, reject, json);
+        });
+      }
```
Comment (dnetrebenko-smartling) [inline middleware/auth-utils/grant-manager.js]: why we couldn't  have one shared handler for `end` action?
```
response.on('end', () => {
    if (response.statusCode < 200 || response.statusCode > 299) {
        const grantType = typeof params === 'object' ? (params.grant_type || '') : '';
        const req = `${options.method}:${options.path} ${grantType}`;
        const resp = `${response.statusCode}: ${http.STATUS_CODES[response.statusCode]} ${json}`;
        reject(new Error(resp + ' on ' + req));
    } else {
        handler(resolve, reject, json);
    }
})
```
?

Comment (a-shevel) [reply middleware/auth-utils/grant-manager.js]: we can have. we will have :)

**Review by dnetrebenko-smartling — APPROVED** [review]

**Pull request merged**